### PR TITLE
Adding a License to the repository

### PR DESCRIPTION
**Pull request by simonsan**
I saw you wrote `MIT` into the Cargo.toml, but that is probably not enough. I added the license files to the repository and stated it in the readme.

**One important thing:** I would really recommend you to use `AGPL` as it's a copyleft license as your work is outstanding and if it's modified and maybe optimized by multi-billion dollar companies in the future, they would need to publish the modified version with it. Otherwise they can just sell your work and their stuff on top of it and earn endless money with it.

> It's conventional, if you don't have a strong reason to do otherwise, to use the same license commonly used in the community you're contributing to. Most Rust crates use same the MIT/Apache-2.0 dual-license as Rust (chosen because it has the patent protections of Apache-2.0 but remains compatible with GPLv2).

> However, I absolutely think that interesting crates should consider using copyleft licenses such as GPLv3, and there's absolutely nothing wrong with doing so. By all means please do, especially if you have a crate that wouldn't be trivial for someone to recreate.
[`Josh Triplett (Rust-lang team, Cargo team)`](https://www.reddit.com/r/rust/comments/jk5g20/eli5_why_do_so_less_people_use_agplv3_for_their/gaijn77/?context=3)

Otherwise this page can help you in choosing a license:
https://choosealicense.com/licenses/agpl-3.0/#

Cheers,
Simon

### CODE REVIEW

**Comment by ronniec95** [comment]
Hi @simonsan 
Thanks for this. I'll merge now.

 I know potentially billion dollar companies will go off and make money on this but I have no real way of stopping them even with a license change to be honest. It's not as if I have access to expensive lawyers to pursue them; I'm just a lowly software developer.

I had generally stayed away from GPL style licences, but this AGPL looks like it will give me acceditation... of course it would be even nicer if they paid me a little bit of those multi-billions :)

**Comment by simonsan** [comment]
@ronniec95 One thing still for you to know, everything important to do if you see a license violation is collected on this page:
https://www.gnu.org/licenses/gpl-violation.html

Cheers!

**Comment by simonsan** [comment]
@ronniec95 You have overwritten the commit with https://github.com/ronniec95/black_scholes/commit/7bcbc393f62ff1ac23c483fef4177461500a9043 , hope you are aware ;)
